### PR TITLE
support PUT with empty body parameter

### PR DESCRIPTION
**Pull request by pnepywoda**
follow-up commit to https://github.com/Netflix/feign/pull/271

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 * Add support for RxJava Observable and Single return types via the `HystrixFeign` builder.
 * Adds fallback implementation configuration to the `HystrixFeign` builder
 * Bumps dependency versions, most notably Gson 2.5 and OkHttp 2.7
+* Supports PUT without a body parameter
```
Comment (adriancole) [inline CHANGELOG.md]: 8.14 has left the building! 8.15 :)

**Comment by adriancole** [comment]
PUT with no content is semantically weird, but yeah understood.

LGTM after filing into 8.15 in the changelog

**Comment by adriancole** [comment]
@pnepywoda mind squashing into one commit? easier for me to click merge vs cherry-pick and squash on my end

**Comment by adriancole** [comment]
thx! let me know if a release is urgent

**Comment by pnepywoda** [comment]
We are waiting on this to get released...if it could happen this week it
would be great!

**Comment by pnepywoda** [comment]
@adriancole actually it's become urgent on our end. Any possibility of release soon?

**Comment by adriancole** [comment]
Yep. Im in transit.

Basically this needs to be cherry picked into 8.x branch and a tag pushed.
I'm embarrassingly without a backup else I'd ask someone else to do this.

@spencergibb I doubt you have access, but would you be a backup for feign?
If so, we can get wheels turning on that.

**Comment by spencergibb** [comment]
I'd be happy to be the backup, though I haven't returned from surgery leave yet.

**Comment by adriancole** [comment]
haha ok so you might be a poor choice of backup at the moment :) Hope you
well on recovery!

**Comment by adriancole** [comment]
8.14.2 on the way

**Pull request merged**